compiler/rustc_mir/src/transform/mod.rs

```diff
@@ -315,6 +317,63 @@ fn mir_promoted(
     (tcx.alloc_steal_mir(body), tcx.alloc_steal_promoted(promoted))
 }
 
+fn mir_for_ctfe<'tcx>(tcx: TyCtxt<'tcx>, def_id: DefId) -> &'tcx Body<'tcx> {
```

Comment:
Oh no, even more confusing functions in this file.^^ Can you add some doc comments to clarify things a bit? In particular that "of const arg" stuff, I have no idea at all what that might be for.

Reply:
https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/ty/struct.WithOptConstParam.html

I already received some questions and know that it confuses some people, so if there is something I can add to the docs to make this clearer I would love to do so

Reply:
Hm, I'm afraid that doc is full of terminology that I do not understand. :/

> A DefId which is potentially bundled with its corresponding generic parameter in case did is a const argument.

What does it mean for did to "be a const argument"? I though DefId refer to definitions, so things like `fn` or `struct` or so?

> Luckily we only need to deal with const arguments once we know their corresponding parameters. 

Arguments and parameters are usually used synonymously I think, except in a few very specific cases around inductive type families in type theory. So I do not know what this sentence means.

Reply:
Actually let me move this to Zulip...

Reply:
Independent of that, ideally there would be doc comments in this file that make sense to a reader that is not already deeply familiar with such concepts -- short comments like this:

https://github.com/rust-lang/rust/blob/13e63f2922905599466dab16ba7e5223b64f360e/compiler/rustc_mir/src/transform/mod.rs#L243